storage/migration.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright © 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+// MigrationVersion represents a version of the database.
+type MigrationVersion int
+
+// MigrationStep represents an action performed to either increase
+// or decrease the migration version of the database.
+type MigrationStep func(tx *sql.Tx) error
+
+// Migration type describes a single Migration.
+type Migration struct {
+	StepUp   MigrationStep
+	StepDown MigrationStep
+}
+
+// migrations is a list of migrations that, when applied in their order,
+// create the most recent version of the database from scratch.
+var migrations []Migration
+
+// AddMigration adds a new migration to the list of available migrations.
+func AddMigration(m Migration) {
+	migrations = append(migrations, m)
+}
+
+// InitMigrationInfo creates a table containing migration information in the database.
+func InitMigrationInfo(db *DBStorage) error {
```

Comment:
I think will be nice to use this function in case if table `migration_info` doesn't exists

Reply:
It would, perhaps, be easier to `CREATE TABLE IF NOT EXISTS` and run that during initialization of the application. WDYT? The second query would be only executed if the first one changed anything. I believe it's possible to check that.